parser/src/asm_ast.rs

```diff
@@ -2,6 +2,52 @@ use number::AbstractNumberType;
 
 use super::ast::{Expression, SelectedExpressions, Statement};
 
+pub mod batched {
```

Comment:
Maybe the parser is not the right place, since this analyzes semantics

Reply:
Right so by an analysis structure do you mean something like

```
struct ASMAnalysis {
   statements: Vec<ASMStatement>,
   batches: Option<Batches>
}
```

where the batcher sets the `batches` member and the pil converter optionally uses the batches?